File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, nums: List[int]) -> int:
        i = 0
        j = len(nums) - 1
        maxArea = 0
        while i < j:
            maxArea = max(maxArea,abs(i - j)*min(nums[i],nums[j]))
            if nums[i] <= nums[j]:
                i += 1
            else:
                j -= 1

        return maxArea

File: Data Structures & Algorithms/balanced-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    bool isTrue = true;
    bool isBalanced(TreeNode* root) {
        if(!root) return true;
        int left = dfs(root->left);
        int right = dfs(root->right);
        if (abs(left - right) > 1) isTrue = false;
        return isTrue;
    }
    int dfs(TreeNode*root){
        if(!root) return 0;
        int left = dfs(root->left);
        int right = dfs(root->right);
        return 1 + max(left,right);
    }
};
